test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -3232,3 +3235,82 @@ UCS_TEST_P(test_ucp_sockaddr_protocols_err_sender,
 UCP_INSTANTIATE_CM_TEST_CASE(test_ucp_sockaddr_protocols_err_sender)
 UCP_INSTANTIATE_TEST_CASE_TLS_GPU_AWARE(test_ucp_sockaddr_protocols_err_sender,
                                         rc_no_ud, "rc_mlx5,rc_verbs")
+
+
+class test_ucp_sockaddr_iface_activate : public test_ucp_sockaddr {
+public:
+    static void
+    get_test_variants(std::vector<ucp_test_variant>& variants)
+    {
+        get_test_variants_mt(variants, UCP_FEATURE_TAG,
+                             CONN_REQ_TAG | TEST_MODIFIER_CM_USE_ALL_DEVICES,
+                             "tag");
+    }
+
+    virtual int is_skip_interface(const struct ifaddrs *ifa) const
+    {
+        return test_ucp_sockaddr::is_skip_interface(ifa) ||
+               ucs::is_rdmacm_netdev(ifa->ifa_name);
+    }
+
+    template <typename Func>
+    void wait_progress(Func func)
+    {
+        ucs_time_t deadline = ucs::get_deadline();
+        scoped_log_handler slh(detect_error_logger);
+        while (!func() && (ucs_get_time() < deadline)) {
+            progress();
+        };
+    }
+
+    ucp_worker_iface_t *get_test_iface(entity &e) const
```

Comment:
IMO we can check if any iface is activated, not check only the one we connect to. because we have TEST_MODIFIER_CM_USE_ALL_DEVICES, other devices can be used as well.

Reply:
This does not work like that, at least on the receiver side, just tested.
```
[     INFO ] receiver after connect:                                                                                                                                                                                               
[     INFO ] iface 0: ens10f0 acount: 0                                                                                                                                                                                                    
[     INFO ] iface 1: ib0 acount: 0                                                                                                                                                                                                        
[     INFO ] iface 2: ib1 acount: 0                                                                                                                                                                                                        
[     INFO ] iface 3: lo acount: 1                                                                                                                                                                                                         
```
activate_count gets incremented only on the interface we connect to.

Reply:
i think in some tests it may work like that, anyway seems more fail-safe to me

Reply:
@iyastreb seems this comment was not addressed , WDYT?

Reply:
I think it is better to check activate count globally and not for a specific device since it will make code simpler and also cover more cases

Reply:
Ok, I implemented that
For that to work I had to explicitly disconnect both client and receiver EPs, while previously I was only disconnecting on the client side. With receiver disconnect we can check that all interfaces are deactivated.
(This updated test still reproduces the issue on the branch without fix, so we are covered here)

Reply:
> Ok, I implemented that For that to work I had to explicitly disconnect both client and receiver EPs, while previously I was only disconnecting on the client side. With receiver disconnect we can check that all interfaces are deactivated. (This updated test still reproduces the issue on the branch without fix, so we are covered here)

Seems testing this discovered an existing issue that activate_count is not back at 0 when client disconnects. However I guess we don't have to fix it in this PR , can add TODO/FIXME note in the test

Reply:
Yes, this activation logic is complex and there are many case and combinations, that I discovered when reproducing this particular issue. I'm sure it deserves a separate task to validate that this feature works as expected. Which is out of scope of this issue. For now I add a TODO note